Android_Assignment_Activity/app/src/main/java/com/weboniselab/meghana/android/assignment5/android_assignment_activity/AppHomePage.java

```diff
@@ -0,0 +1,53 @@
+package com.weboniselab.meghana.android.assignment5.android_assignment_activity;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.Button;
+import android.widget.TextView;
+
+/**
+ * Created by webonise on 10/8/15.
+ */
+public class AppHomePage extends Activity{
+    private final String TAG=getClass().getSimpleName();
+    private String msg;
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main2);
+        Log.v(TAG, "***On Create***");
+        Intent intent=getIntent();
+        Bundle bundle=intent.getExtras();
+        msg=bundle.getString("message");
```

Comment:
From Next time all Intent Keys should be declared as constants. No hardcoded strings